packages/SwingSet/src/kernel/kernel.js

```diff
@@ -706,14 +706,14 @@ export default function buildKernel(
     vatParameters = {},
     creationOptions = {},
   ) {
-    if (typeof setup !== 'function') {
-      throw Error(`setup is not a function, rather ${setup}`);
-    }
+    assert.typeof(
+      setup,
+      'function',
+      d`setup is not a function, rather ${setup}`,
+    );
     assertKnownOptions(creationOptions, ['enablePipelining', 'metered']);
 
-    if (kernelKeeper.hasVatWithName(name)) {
-      throw Error(`vat ${name} already exists`);
-    }
+    assert(!kernelKeeper.hasVatWithName(name), d`vat ${name} already exists`);
```

Comment:
This is a reasonable example of the kind of thing I'm talking about when I complain about inverting the condition.
The pattern I like is ```
if (badThing) {
  throw Error(`it's very bad`);
}
```
rather than
```
assert(!badThing, `it's very bad`);
```
I think the key difference between our intuitions is that you're thinking "make sure there's not a bad thing before continuing", whereas I'm thinking "look for a bad thing and if found act upon it".

Reply:
I'd remove the double negative from the characterization of my thinking: "Make sure this condition is true before continuing, because the following code assumes that it is true."